### PR TITLE
Client Report Metrics

### PR DESCRIPTION
**Pull request by kimyen**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/dashboard/dao/redis/NameSpace.java [hunk 1]
```diff
@@ -27,8 +27,5 @@
     bitcount,
 
     /** Simple counts. */
-    count,
-
-    /** File Download Report */
-    fdownload
+    count
```
Comment (kimyen) [inline src/main/java/org/sagebionetworks/dashboard/dao/redis/NameSpace.java]: never used fdownload namespace

File: src/main/java/org/sagebionetworks/dashboard/parse/UserDataReader.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
     public String read(AccessRecord record) {
         Long timestamp = record.getTimestamp();
         String userId = (new UserIdReader()).read(record);
-        String client = (new ClientSummaryReader()).read(record);
+        String client = (new ClientReader()).read(record);
```
Comment (kimyen) [inline src/main/java/org/sagebionetworks/dashboard/parse/UserDataReader.java]: show details about the client

File: src/main/java/org/sagebionetworks/dashboard/service/MetricReader.java [hunk 3]
```diff
@@ -102,15 +102,15 @@
...
+    public List<UserDataPoint> getAllReport(String metricName, String input) {
         if (metricName == null || metricName.isEmpty()) {
             throw new IllegalArgumentException("Metric name cannot be null or empty.");
         }
         String metricId = getMetricId(metricName);
-        if (entityId.startsWith("syn")) {
-            entityId = entityId.substring(3);
+        if (input.startsWith("syn")) {
+            input = input.substring(3);
         }
```
Comment (kimyen) [inline src/main/java/org/sagebionetworks/dashboard/service/MetricReader.java]: input could be an entityId (for file download report) or a client version

File: src/test/java/org/sagebionetworks/dashboard/dao/redis/KeyCachedDaoImplTest.java [hunk 4]
```diff
@@ -29,10 +29,10 @@
...
     private static final String failMetric = "questionFailMetric";
     private static final String prefix = "n:month:uniquecount:";
-    private static final CuResponseRecord pass1 = new CuResponseRecord(1, 1, new DateTime(1403827200000L), true);
-    private static final CuResponseRecord pass2 = new CuResponseRecord(12, 1, new DateTime(1403827200001L), true);
-    private static final CuResponseRecord fail1 = new CuResponseRecord(1, 2, new DateTime(1403827200000L), false);
-    private static final CuResponseRecord fail2 = new CuResponseRecord(12, 2, new DateTime(153827200001L), false);
+    private static final CuResponseRecord pass1 = new CuResponseRecord(1, 1, new DateTime(2014, 06, 27, 0, 0), true);
+    private static final CuResponseRecord pass2 = new CuResponseRecord(12, 1, new DateTime(2014, 06, 27, 0, 1), true);
+    private static final CuResponseRecord fail1 = new CuResponseRecord(1, 2, new DateTime(2014, 06, 27, 0, 0), false);
+    private static final CuResponseRecord fail2 = new CuResponseRecord(12, 2, new DateTime(2014, 07, 27, 0, 1), false);
```
Comment (kimyen) [inline src/test/java/org/sagebionetworks/dashboard/dao/redis/KeyCachedDaoImplTest.java]: creating a new Date with a long timestamp cause the problem on the edge cases for floorToMonth. 6/1/2014 12AM wont be recognize as the 1st day of June 2014.
